extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/sql/resources/SqlStatementResource.java

```diff
@@ -150,16 +152,12 @@ public Response doPost(final SqlQuery sqlQuery, @Context final HttpServletReques
       );
       if (ExecutionMode.ASYNC != executionMode) {
         return buildNonOkResponse(
-            DruidException.forPersona(DruidException.Persona.USER)
-                          .ofCategory(DruidException.Category.INVALID_INPUT)
-                          .build(
-                              StringUtils.format(
-                                  "The statement sql api only supports sync mode[%s]. Please set context parameter [%s=%s] in the context payload",
-                                  ExecutionMode.ASYNC,
-                                  QueryContexts.CTX_EXECUTION_MODE,
-                                  ExecutionMode.ASYNC
-                              )
-                          )
+            InvalidInput.exception(
+                "The statement sql api only supports sync mode[%s]. Please set context parameter [%s=%s] in the context payload",
```

Comment:
I'm not sure I know what the `statement sql api` is?  If it's only supposed to be one thing and this code  knows what that one thing is and the user got their request to this line of the code, why does the user need to change their query to only do the one thing that this code can do?  I.e. why not just ignore the context parameter entirely?  Or overwrite it with the one value that this code can do?

Reply:
Since the sql statement api will support the "SYNC" mode soon, I wanted the user to set the flag explicitly. 

Rolling upgrades then become easy since we donot have a default mode/overridden mode. 

The mode also changes the way the results are fetched. 
If the mode set is async, the results are fetched via the results api. 
If the mode set is 'sync', then the results are streamed back in the same post request. (At-least, that's how I envision it. )

Reply:
I went through the code, and ExecutionMode is currently unused. I have left it as is since I am not aware if it will be utilized later, however, I changed the exception to defensive since this parameter isn't supported and unset in the first place.

(cc @cryptoe, I can clean this up in case we can remove the class)